libs/game/tilemap.ts

```diff
@@ -315,6 +315,32 @@ namespace tiles {
             return output;
         }
 
+        public sampleTilesByType(index: number, maxCount: number): Location[] {
+            if (this.isInvalidIndex(index) || !this.enabled || maxCount <= 0) return [];
+
+            let count = 0;
+            const reservoir: Location[] = [];
+            for (let col = 0; col < this._map.width; ++col) {
+                for (let row = 0; row < this._map.height; ++row) {
+                    let currTile = this._map.getTile(col, row);
+                    if (currTile === index) {
+                        // first **maxCount** elements just enqueue
+                        if (count < maxCount) {
+                            reservoir.push(new Location(col, row, this));
```

Comment:
Hmm, @jwunderl  please correct me if I'm reading this wrong but it looks like if (0,0) matches the tile type you need, it'll always be in the reservoir and if the reservoir is size 1 it'll always be returned. Is that right?

Reply:
ooh I see how this works now. Nvm